tests/test_tpa_pipeline.py

```diff
@@ -155,6 +155,33 @@ def test_handle_enterprise_logistration_consent_not_required(self):
                 enterprise_customer=enterprise_customer,
                 user_id=self.user.id
             ).count() == 1
+            assert UserDataSharingConsentAudit.objects.filter(
+                user__user_id=self.user.id,
+                user__enterprise_customer=enterprise_customer,
+            ).count() == 0
+
+    def test_handle_enterprise_logistration_consent_externally_managed(self):
+        """
+        Test that when consent is externally managed, we create an EnterpriseCustomerUser and
+        UserDataSharingConsentAudit object, then return.
+        """
+        backend = mock.MagicMock(name=None)
```

Comment:
This doesn't do what you would probably think it does. Passing `name` as an arg to the mock instantiator does _not_ cause it to mock out the `name` attribute on the produced mock. You have to create the mock and then do `mock.name = None`.

```python
>>> import mock
>>> x = mock.MagicMock(name=None)
>>> x
<MagicMock id='4548168976'>
>>> print(x.name)
<MagicMock name='mock.name' id='4548479440'>
>>> x.name = None
>>> print(x.name)
None
```

Are you using that value to test anything? If so, double-check your tests, because they're not working. If not, then we can probably skip putting a value there anyway - mocks will automatically return another `MagicMock` for any attributes that aren't explicitly defined.